docs/DeveloperGuide.md

```diff
@@ -355,62 +355,108 @@ We have chosen to implement the `delete` command to accept the index of the pers
+After which, the `extractValidKeyword` method ensures that the keyword provided in the input is valid in the case that the prefix is `mt|` or `lt|`, 
+since these two prefixes specifically require a numerical value as the keyword instead of a string value. 
 
-##### Predicate Creation
+##### Executing the Command
 
-The `PersonDetailContainsKeywordPredicate` class implements the `Predicate` interface to filter contacts based on search criteria. It takes a prefix and keyword as parameters, allowing it to filter contacts based on specific details like name, phone number, etc.
+The `FindCommand` class is responsible for executing the command for filtering the list in the application. 
 
-With the prefix and the value extracted from parsing the user input, a `PersonDetailContainsKeywordPredicate` is created.
+Using the prefix and keyword from parsing user input, a `FindCommand` is created. the `execute` method is then called by the `LogicManager`.
 
-##### Executing the Command
+**Creating Predicate** <br>
 
-The `FindCommand` class is responsible for executing the command for filtering the list in the application. It takes in a `PersonDetailContainsKeywordPredicate` as a parameter and has a `execute` method inherited from its parent class of `Command`
+<box type="info" seamless>
 
-Using the `PersonDetailContainsKeywordPredicate` created from parsing user input, a `FindCommand` is created. the `execute` method is then called by the `LogicManager`.
+**Note:** The `PersonDetailPredicate` and `ExamPredicate` classes implement the `Predicate` interface to filter contacts based on the search criteria.
+A brief overview of the two classes is given below:
+* `PersonDetailPredicate` takes a prefix and keyword as parameters, allowing it to filter contacts based on specific details like name, phone number, etc.
+* `ExamPredicate` takes a prefix, a keyword, and an exam as parameters, allowing it to filter contacts based on exam scores of a specific exam.
+</box>
+
+
+The find command first checks if an exam is required by checking if the prefix is `mt|` or `lt|`. 
+If an exam is required, the `selectedExam` is retrieved from the `model` and passed to the `ExamPredicate` constructor along with the prefix and keyword.
+Otherwise, the `PersonDetailPredicate` class is created with the prefix and keyword.
 
 **Updating Filtered Person List** <br>
 
-The `ModelManager` class implements the `Model` interface and manages the application's data. It maintains a `filteredPersons` list, which is a FilteredList of contacts based on the applied predicate. The `updateFilteredPersonList` method implemented in `ModelManager` updates the filtered list based on the provided predicate.
+The `ModelManager` class implements the `Model` interface and manages the application's data. It maintains a `filteredPersons` list, 
+which is a FilteredList of contacts based on the applied predicate. The `updateFilteredPersonList` method implemented in `ModelManager` 
```

Comment:
may want to edit FilteredList to filtered list? unless it is a class then maybe can put `FilteredList`?

Reply:
rectified! good spot!